### PR TITLE
WebGLRenderer: Fix transmission render target type check

### PR DESCRIPTION
**Pull request by chubei-oppen**
**Description**

Currently, transmission won't work under

1. A WebGL2 context without either `EXT_color_buffer_half_float` or `EXT_color_buffer_float` support.
2. A WebGL1 context without `EXT_color_buffer_half_float` support.

This commit fixes that.

*This contribution is funded by [OppenFuture Technologies](https://oppentech.com/en)*

### CODE REVIEW

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -1196,12 +1196,13 @@ function WebGLRenderer( parameters = {} ) {
 	function renderTransmissionPass( opaqueObjects, scene, camera ) {
 
 		const isWebGL2 = capabilities.isWebGL2;
+		const hasHalfFloatColorBuffer = extensions.has( 'EXT_color_buffer_half_float' ) || ( isWebGL2 && extensions.has( 'EXT_color_buffer_float' ) );
```
Comment (Mugen87) [inline src/renderers/WebGLRenderer.js:1199]: Isn't it sufficient to just test for `EXT_color_buffer_half_float`? 

I don't think there is a WebGL 2 device that does `EXT_color_buffer_float` but not `EXT_color_buffer_half_float`.

File: src/renderers/WebGLRenderer.js [hunk 2]
```diff
@@ -1196,12 +1196,13 @@ function WebGLRenderer( parameters = {} ) {
...
 
 		if ( _transmissionRenderTarget === null ) {
 
 			_transmissionRenderTarget = new WebGLRenderTarget( 1, 1, {
 				generateMipmaps: true,
-				type: utils.convert( HalfFloatType ) !== null ? HalfFloatType : UnsignedByteType,
+				type: hasHalfFloatColorBuffer ? HalfFloatType : UnsignedByteType,
```
[the remark this answers is not in the document]
Comment (chubei-oppen) [reply src/renderers/WebGLRenderer.js:1205]: The `EXT_color_buffer_half_float` [specification](https://www.khronos.org/registry/webgl/extensions/EXT_color_buffer_half_float/) _seems_ to imply that `OES_texture_half_float` must be supported by stating `These and textures created with type = HALF_FLOAT_OES`. Of course adding that check won't hurt.

File: src/renderers/WebGLRenderer.js (hunk 1 above)
Comment (chubei-oppen) [reply src/renderers/WebGLRenderer.js:1199]: Yes no reasonable device would do that, but the `EXT_color_buffer_float` [specification](https://www.khronos.org/registry/webgl/extensions/EXT_color_buffer_float/) says `This is a difference in functionality compared to the [EXT_color_buffer_half_float](https://www.khronos.org/registry/webgl/extensions/EXT_color_buffer_half_float) extension`, so I guess a real evil guy can do it.

Comment (Mugen87) [reply src/renderers/WebGLRenderer.js:1199]: I think I would take this risk and go for simplicity. Meaning only checking for `EXT_color_buffer_half_float`.

File: src/renderers/WebGLRenderer.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mugen87) [reply src/renderers/WebGLRenderer.js:1205]: In this case, let's keep the current implementation. If someone complains, we can add the check later, too.

File: src/renderers/WebGLRenderer.js (hunk 1 above)
Comment (chubei-oppen) [reply src/renderers/WebGLRenderer.js]: Commit updated. Please note that https://github.com/mrdoob/three.js/blob/8db7a598ed6e71538e286e5c56c02cc5823157dc/src/renderers/WebGLRenderer.js#L1976 is also doing the `EXT_color_buffer_float` check, so there's a little inconsistency.

**Review by Mugen87 — APPROVED** [review]

File: src/renderers/WebGLRenderer.js (hunk 1 above)
Comment (Mugen87) [reply src/renderers/WebGLRenderer.js]: If the approach of this PR works fine, we can refactor this code with a later PR.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**